### DISCUSSION TITLE
Laravel-echo : Customizing The Authorization Request

### DISCUSSION THREAD

**Discussion by ParzivalLashly** [laravel/framework#42946] [Q&A]
hi, im using **Laravel 8** for api backend and **Reactjs** for frontend, **Laravel echo@1.11.7** and **Laravel echo server** with **socket.io-client** for ws connections between front and backend
for some reasons i need to authenticate private channels with custom route and controller.
problem is Laravel echo authorizer not working for me ( actually nothing happend ) .

```
window.Echo = new Echo({
    // ...
    authorizer: (channel, options) => {
        return {
            authorize: (socketId, callback) => {
                axios.post('/api/broadcasting/auth', {
                    socket_id: socketId,
                    channel_name: channel.name
                })
                .then(response => {
                    callback(false, response.data);
                })
                .catch(error => {
                    callback(true, error);
                });
            }
        };
    },
})
```
any suggestion ?

**Comment by erikn69** [comment]
https://github.com/laravel/framework/issues/23268#issuecomment-443986055
https://github.com/laravel/framework/issues/23268#issuecomment-542890314
